clients/src/main/java/org/apache/kafka/clients/GroupRebalanceConfig.java

```diff
@@ -73,6 +74,16 @@ public GroupRebalanceConfig(AbstractConfig config, ProtocolType protocolType) {
             this.groupInstanceId = Optional.empty();
         }
 
+        // The WorkerGroupMember in connect module also uses this class, but there is no client.rack in DistributedConfig.
+        // Ignore the rackId in that case to avoid ConfigException.
+        // The GroupCoordinatorService throws error if the rackId is empty. The default value of client.rack is empty string.
+        // Skip empty rackId to avoid InvalidRequestException.
+        if (config.values().containsKey(CommonClientConfigs.CLIENT_RACK_CONFIG) && !config.getString(CommonClientConfigs.CLIENT_RACK_CONFIG).isEmpty()) {
+            this.rackId = Optional.ofNullable(config.getString(CommonClientConfigs.CLIENT_RACK_CONFIG));
+        } else {
+            this.rackId = Optional.empty();
+        }
```

Comment:
- Should we only do this if the protocolType is CONSUMER?
- It may be simpler to get the config in a local variable and to set this.rack if non-null and non-empty;